recordedit/recordEdit.app.js

```diff
@@ -341,23 +377,42 @@
                             }
 
                             recordEditModel.rows[0][column.name] = initialModelValue;
-                        };
+                        }
 
                         $rootScope.displayReady = true;
                         // if there is a session, user isn't allowed to create
                     } else if (session) {
-                        var forbiddenError = new ERMrest.ForbiddenError(messageMap.unauthorizedErrorCode, messageMap.unauthorizedMessage);
-                        throw forbiddenError;
+                        throw new ERMrest.ForbiddenError(messageMap.unauthorizedErrorCode, messageMap.unauthorizedMessage);
                         // user isn't logged in and needs permissions to create
                     } else {
-                        var notAuthorizedError = new ERMrest.UnauthorizedError(messageMap.unauthorizedErrorCode, messageMap.unauthorizedMessage);
-                        throw notAuthorizedError;
+                        throw new ERMrest.UnauthorizedError(messageMap.unauthorizedErrorCode, messageMap.unauthorizedMessage);
                     }
                 }
             }, function error(response) {
                 throw response;
             });
         });
 
+
+        function getForeignKeyData (rowIndex, colName, fkName, fkRef) {
```

Comment:
Does this work defining the function here? I would imagine it needs to be defined above where it's used in the script. But maybe because it's in a callback, it's fine here.

Reply:
It does work. It's inside the run block, and I could move it inside the resolve callback. But I thought that it is going to make it more difficult to read.